File: 209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int l = 0, r = 0;
        int res = nums.size() + 1;
        while (r < nums.size())
        {
            int sum = 0;
            for (int i = l; i <= r; i++)
                sum += nums[i];
            if (sum < s) r++;
            else
            {
                res = res < r+1-l ? res : r+1-l;
                l++;
            }
        }
        return res == nums.size() + 1 ? 0 : res;
    }
};

File: 209. Minimum Size Subarray Sum.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        l = r = 0
        res = len(nums) + 1
        while r < len(nums):
            if sum(nums[l:r+1]) < s:
                r += 1
            else:
                res = min(res, r+1-l)
                l += 1
        return res if res < len(nums) + 1 else 0
